### ISSUE TITLE
Training error

### ISSUE THREAD

**Issue by maikelronnau** [rdcolema/tensorflow-image-classification#2]
I downloaded the kaggle dogs and cast dataset to run the training process in the same conditions as the tutorial, but I'm getting this error when running the code:

Reading training images
Loading dogs files (Index: 0)
Loading cats files (Index: 1)
Reading test images
Size of:
- Training-set:		20000
- Test-set:		12500
- Validation-set:	5000

Epoch 1 --- Training Accuracy:  56.2%, Validation Accuracy:  56.2%, Validation Loss: 0.682
Time elapsed: 0:00:01
Traceback (most recent call last):
  File "cnn.py", line 787, in <module>
    print_validation_accuracy()
  File "cnn.py", line 743, in print_validation_accuracy
    feed_dict = {x: images,
UnboundLocalError: local variable 'x' referenced before assignment

I've tried using "global x" to make sure it was referencing to the correct value, but then I got this error:

Reading training images
Loading dogs files (Index: 0)
Loading cats files (Index: 1)
Reading test images
Size of:
- Training-set:		20000
- Test-set:		12500
- Validation-set:	5000
Epoch 1 --- Training Accuracy:  62.5%, Validation Accuracy:  43.8%, Validation Loss: 0.703
Time elapsed: 0:00:01
Traceback (most recent call last):
  File "cnn.py", line 788, in <module>
    print_validation_accuracy()
  File "cnn.py", line 736, in print_validation_accuracy
    images = data.valid.images[i:j, :].reshape(batch_size, img_size_flat)
ValueError: cannot reshape array of size 393216 into shape (16,49152)

I also tried checking the code with the original tutorial in with this was based, but still wasn't able to solve the problem and train the model.

What should I do to solve that? Thanks

**Comment by rdcolema** [comment]
Hey @maikelronnau, you're running this as a single python script and not as a notebook right (judging from the traceback)? I'm guessing that's affecting variable scopes and causing the first error. 

The second error looks like a batch size mismatch. I believe the issue here is that your validation split is different, and is no longer evenly divisible by the batch size: 

`5000/16 = 312.5` in your example
 vs
`4000/16 = 250` in my example

If you're still unable to resolve, you might be able to just continue without printing out the validation accuracy, since it looks like training itself is working fine.

**Comment by maikelronnau** [comment]
Hi!

Sorry for the late response, I wish I could've posted it before but I'm very busy with the semester here.

Yes, I'm running a Python script I've converted using Jupyter's provided method. I changed the validation size to .24 to correspond my training size (25000, and not 20000 as in your tutorial. I just assumed mine was 20000 too), now I'm getting this message:

Reading training images
Loading dogs files (Index: 0)
Loading cats files (Index: 1)
Reading test images
Size of:
- Training-set:		19000
- Test-set:		12500
- Validation-set:	6000
Epoch 1 --- Training Accuracy:  62.5%, Validation Accuracy:  43.8%, Validation Loss: 0.731
Time elapsed: 0:00:01
Accuracy on Test-Set: 50.5% (3033 / 6000)
Traceback (most recent call last):
  File "cnn.py", line 797, in <module>
    optimize(num_iterations=99)  # We already performed 1 iteration above.
  File "cnn.py", line 605, in optimize
    session.run(optimizer, feed_dict=feed_dict_train)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 767, in run
    run_metadata_ptr)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 922, in _run
    + e.args[0])
TypeError: Cannot interpret feed_dict key as Tensor: Can not convert a int64 into a Tensor.

So I commented the code for printing the validation accuracy but the message still happen. I will debug the code and try to fix the problem. 

Thank you very much!
